### PR TITLE
feat: add jwt authorization to supply chain example

### LINKED ISSUE #1579: docs(examples): add authN/authZ for supply chain app example
### Description

We want to be reasonably sure that we haven't forgotten anything basic from the JWT support code in the API server and the best way to find out is to try and make one of the end to end examples work with JWT authorization so that we can put ourselves in the shoves of someone who is developing an application with the framework leveraging the plugins and the API server.
While doing the above, likely some issues will come to light that otherwise would've been hidden until we go to 1.0.0 and people start using it so this way we can get ahead of that.
If any issues arise, please make sure to create new, separate tasks for fixing those that can also be addressed by separate PRs so that we avoid one huge PR that updates the examples and also updates core/API-server/core-api package code at the same time.
By the end of it in the PR resolving this issue we should only have the changes that are touching only the examples folder code because everything else necessary (prerequisites) was done already.

### Acceptance Criteria
1. The supply chain example is locked down by default with authorization, e.g. JWT validation is enabled
2. To keep it user friendly the example tutorials need to end with valid JWTs printed to the command line at the very end so that end users can see/use them with the least amount of hassle possible
3. Do not mix changes between examples and core code (further explained above but wanted to underline it here too)
4. Make sure the documentation is updated accordingly as well
5. Publish the container image to GHCR and also make sure to update the root README.md and the inner supply-chain-app README.md with those images.

### PR DESCRIPTION
**Pull request by elenaizaguirre**
closes #1579

Signed-off-by: Elena Izaguirre <e.izaguirre.equiza@accenture.com>

### CODE REVIEW

**Review by takeutak — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -34,9 +34,10 @@ As blockchain technology proliferates, blockchain integration will become an inc
       -p 4200:4200 \
       ghcr.io/hyperledger/cactus-example-supply-chain-app:2021-09-08--docs-1312
     ```
-3. Wait for the output to show the message `INFO (api-server): Cactus Cockpit reachable http://0.0.0.0:3100`
-4. Visit http://localhost:3100 in a web browser with Javascript enabled
-5. Use the graphical user interface to create data on both ledgers and observe that a consistent view of the data from different ledgers is provided.
+3. Wait for the output to show the message `INFO (api-server): Cactus Cockpit reachable http://0.0.0.0:3200`
```
Comment (petermetz) [inline README.md:37]: @elenaizaguirre Reminder/note to self: Once the PR is finalized we need to do an extra step of publishing the new image and then updating the tag in here so that people can get the updated code you wrote (the old one won't have the prompt)

File: examples/supply-chain-app/README.md [hunk 2]
```diff
@@ -61,4 +61,6 @@ On the terminal, issue the following commands (steps 1 to 6) and then perform th
 7. Locate the `.vscode/template.launch.json` file
 8. Within that file locate the entry named `"Example: Supply Chain App"`
 9. Copy the VSCode debug definition object from 2) to your `.vscode/launch.json` file
-10. At this point the VSCode `Run and Debug` panel on the left should have an option also titled `"Example: Supply Chain App"` which
\ No newline at end of file
+10. At this point the VSCode `Run and Debug` panel on the left should have an option also titled `"Example: Supply Chain App"` which starts the application
```
Comment (petermetz) [inline examples/supply-chain-app/README.md:64]: @elenaizaguirre I went through the steps here and then it worked to the point of me getting the prompt, pasing in the JWT, then the get JWS endpoint succeeding, but then the next endpoint call fails. Can you confirm if all the functionality of the web app works on your end? I'm attaching some screenshots, hopefully they help a little.

**Review by petermetz — CHANGES_REQUESTED** [review]

File: examples/supply-chain-app/README.md (hunk 2 above)
Comment (elenaizaguirre) [reply examples/supply-chain-app/README.md:64]: @petermetz I have cleaned up my environment and compiled the code again to check that I was not using old compiled code but always works for me.

tempsnip
Captura

Comment (elenaizaguirre) [reply examples/supply-chain-app/README.md:64]: @petermetz 
I have run it on a newly installed virtual machine (Ubuntu 20.04) and it still works for me with all functionality working. I don't know what's happening...

Comment (petermetz) [reply examples/supply-chain-app/README.md:64]: @elenaizaguirre I got to the bottom of it and realized that it's my own fault. (Well, I'll blame Chrome but still it was me).
I was visiting the link that the application prints to the terminal which has a host of `0.0.0.0` instead of `localhost` and Chrome is blocking CORS requests on hosts like that apparently (it was working fine if I did it with cURL so I know for a fact that it's Chrome doing it).

And the reason why the JWS request worked and the others didn't is because the JWS request goes through to the same API server that is hosting the reverse proxy+GUI (so no CORS needed) but the API server that has the quorum connector plugin installed is on a different host and the client side does not do URL rewriting on the node host that it gets from the consortium database so it tries to hit the 4xxx ports while the GUI is running on 3xxx and then Chrome shuts that down.

TLDR: LGTM

File: README.md (hunk 1 above)
Comment (petermetz) [reply README.md:37]: @elenaizaguirre It's good to go now, so please publish the new image and then update here and also in the inner README of the supply chain app itself (best to do a free text search for the image tag project-wide to find all usages).
Once that's done, it's ready to merge IMO.

**Review by petermetz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (elenaizaguirre) [reply README.md:37]: @petermetz As we talked about [here](https://github.com/hyperledger/cactus/issues/1579#issuecomment-1022717133), I need the newest npm packages before create the image

**Review by petermetz — CHANGES_REQUESTED** [review]
> @petermetz As we talked about [here](https://github.com/hyperledger/cactus/issues/1579#issuecomment-1022717133), I need the newest npm packages before create the image 

@elenaizaguirre Ah, I'm so sorry, I'm struggling to keep up with everything... :-)
I just published your branch to npm with these package names (pasted below).
Does this make it possible now to build and publish the docker image so that the tag in the readme can also be updated? I'm pushing for this because if we merge this PR as-is then the instructions in the README will get updated, but without the new image the new instructions won't make sense to people so I opted to eliminate the deadlock by publishing the npm packages as a canary release. Sorry again for all this back and forth and the complications. Release management is not easy due to the dependencies unfortunately.

```sh
Successfully published:
 - @hyperledger/cactus-example-carbon-accounting-backend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-example-carbon-accounting-business-logic-plugin@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-example-carbon-accounting-frontend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-example-supply-chain-backend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-example-supply-chain-business-logic-plugin@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-example-supply-chain-frontend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-htlc-coordinator-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-object-store-ipfs@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-api-client@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-cmd-api-server@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-cmd-socket-server@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-common@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-core-api@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-core@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-consortium-manual@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-htlc-eth-besu-erc20@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-htlc-eth-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-keychain-aws-sm@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-keychain-azure-kv@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-keychain-google-sm@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-keychain-memory-wasm@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-keychain-memory@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-keychain-vault@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-corda@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-fabric-socketio@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-fabric@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-go-ethereum-socketio@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-iroha@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-quorum@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-sawtooth-socketio@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-ledger-connector-xdai@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-plugin-odap-hermes@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-api-client@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-cmd-api-server@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-plugin-consortium-manual@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-plugin-htlc-eth-besu-erc20@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-plugin-htlc-eth-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-plugin-ledger-connector-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-plugin-ledger-connector-quorum@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
 - @hyperledger/cactus-test-tooling@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
```

**Comment by elenaizaguirre** [comment]
> > @petermetz As we talked about [here](https://github.com/hyperledger/cactus/issues/1579#issuecomment-1022717133), I need the newest npm packages before create the image
> 
> @elenaizaguirre Ah, I'm so sorry, I'm struggling to keep up with everything... :-) I just published your branch to npm with these package names (pasted below). Does this make it possible now to build and publish the docker image so that the tag in the readme can also be updated? I'm pushing for this because if we merge this PR as-is then the instructions in the README will get updated, but without the new image the new instructions won't make sense to people so I opted to eliminate the deadlock by publishing the npm packages as a canary release. Sorry again for all this back and forth and the complications. Release management is not easy due to the dependencies unfortunately.
> 
> ```shell
> Successfully published:
>  - @hyperledger/cactus-example-carbon-accounting-backend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-example-carbon-accounting-business-logic-plugin@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-example-carbon-accounting-frontend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-example-supply-chain-backend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-example-supply-chain-business-logic-plugin@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-example-supply-chain-frontend@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-htlc-coordinator-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-object-store-ipfs@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-api-client@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-cmd-api-server@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-cmd-socket-server@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-common@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-core-api@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-core@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-consortium-manual@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-htlc-eth-besu-erc20@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-htlc-eth-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-keychain-aws-sm@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-keychain-azure-kv@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-keychain-google-sm@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-keychain-memory-wasm@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-keychain-memory@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-keychain-vault@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-corda@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-fabric-socketio@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-fabric@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-go-ethereum-socketio@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-iroha@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-quorum@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-sawtooth-socketio@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-ledger-connector-xdai@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-plugin-odap-hermes@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-api-client@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-cmd-api-server@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-plugin-consortium-manual@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-plugin-htlc-eth-besu-erc20@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-plugin-htlc-eth-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-plugin-ledger-connector-besu@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-plugin-ledger-connector-quorum@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
>  - @hyperledger/cactus-test-tooling@1.0.0-jwt-supply-chain.62c7e930.0+62c7e930
> ```

@petermetz Don't worry, I know you are very busy and I am very grateful for your help. 

Regarding the npm packages, I think something went wrong. When I install `cactus-example-supply-chain-backend` in Dockerfile setting that new version as the value of NPM_PKG_VERSION, I get an error (code try to validate a key before set it) that I solved in  #1655. 

I checked several package.json of the container but only `cactus-example-supply-chain-backend` has this new version, other packages are installed with the previous version. Could it be caused by needing to do a build before publish?

**Review by petermetz — APPROVED** [review]
@elenaizaguirre Very sorry for the slow turnaround I finally managed to get this over the finish line.
Published the image with the freshly published canary npm packages as `ghcr.io/hyperledger/cactus-example-supply-chain-app:2022-04-05--feat-1579` and updated your PR so that the README files point to this one as well.

LGTM
